src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/simple/RandomizedJoinPlanOptimizerImpl.java

```diff
@@ -0,0 +1,91 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.simple;
+
+import java.util.List;
+import java.util.Random;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.utils.PhysicalPlanFactory;
+import se.liu.ida.hefquin.engine.queryproc.QueryOptimizationException;
+
+public class RandomizedJoinPlanOptimizerImpl extends JoinPlanOptimizerBase {
+	@Override // Implement initializeEnumerationAlgorithm from JoinPlanOptimizerBase.
+	public EnumerationAlgorithm initializeEnumerationAlgorithm( final List<PhysicalPlan> subplans ) {
+		return new RandomizedEnumerationAlgorithm(subplans);
+	}
+	
+	protected class RandomizedEnumerationAlgorithm implements EnumerationAlgorithm {
+		
+		/**
+		 * The class contains a random seed object.
+		 */
+		Random random;
+		
+		/**
+		 * Have a list of subplans.
+		 */
+		protected final List<PhysicalPlan> subplans;
+
+		/**
+		 *  When initializing the algorithm, start with the given list of subplans.
+		 *  @param subplans
+		 */
+		public RandomizedEnumerationAlgorithm( final List<PhysicalPlan> subplans ) {
+			this.subplans = subplans;
+			this.random = new Random(); // Generates the random seed.
+		}
+		
+
+		/**
+		 * Also taken from Greedy and left unchanged save for the different method being called within.
+		 */

```

Comment:
This comment does not make much sense in the context of this class. I understand that you added it as a way of documenting what you have done. However, before merging this PR into the main branch, such comments should be removed. Someone who looks at this code later may not know the context in which you made this comment and, then, the comment might cause confusion.